src/client/texturesource.cpp

```diff
@@ -179,22 +185,30 @@ TextureSource::TextureSource()
 	m_textureinfo_cache.emplace_back(TextureInfo{""});
 	m_name_to_id[""] = 0;
 
-	// Cache some settings
-	// Note: Since this is only done once, the game must be restarted
-	// for these settings to take effect.
+	for (auto *name : {"mip_map", "trilinear_filter", "bilinear_filter",
+		"anisotropic_filter"}) {
+		g_settings->registerChangedCallback(name,
+			&settingChangedCallback, this);
+	}
+	readSettings();
+}
+
+void TextureSource::readSettings()
+{
 	mesh_filter_needed =
-			g_settings->getBool("mip_map") ||
-			g_settings->getBool("trilinear_filter") ||
-			g_settings->getBool("bilinear_filter") ||
-			g_settings->getBool("anisotropic_filter");
+		g_settings->getBool("mip_map") ||
+		g_settings->getBool("trilinear_filter") ||
+		g_settings->getBool("bilinear_filter") ||
+		g_settings->getBool("anisotropic_filter");
```

Comment:
why this whitespace-only change? didn't we have "2 tabs for continuation lines" rule?

Reply:
do we? I always use 1 tab for continuation

Reply:
lol

I guess I was thinking of this:

> In case your function parameters don’t fit within the defined line length, use the following style. Indention for continuation lines is exactly two tabs.
>
> ~ https://dev.luanti.org/engine-dev-process/code-style-guidelines/#function-declarations

other examples in that page also use two tabs, e.g.

```cpp
bool success =
		np_terrain_base  && np_terrain_higher && np_steepness &&
		np_height_select && np_trees          && np_mud       &&
		np_beach         && np_biome          && np_cave;
```
> ~ https://dev.luanti.org/engine-dev-process/code-style-guidelines/#spaces

Reply:
and this one:

> When breaking conditionals, indent following lines of the conditional with two tabs and the statement body with one tab.
>
> ~ https://dev.luanti.org/engine-dev-process/code-style-guidelines/#spaces